### PR TITLE
Dala 4251 fix company owner cant upload document

### PR DESCRIPTION
**Pull request by MichBlank**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [ ] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [x] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [x] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [x] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [x] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt [hunk 1]
```diff
@@ -91,13 +95,63 @@ class DocumentControllerTest {
         assertEquals(HttpStatus.NOT_FOUND.value(), exception.statusCode)
     }
 
+    @Test
+    fun `test that non admin users can upload documents only when they are CompanyOwner and not otherwise`() {
```
Comment (emanueldfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt:99]: Würde die Testbeschreibung leicht anpassen, weil sie nicht ganz der Wahrheit entspricht:
Denn User mit keycloak-uploader Rolle sind ja auch non-admins, können aber trotzdem Dokumente hochladen.

Vorschlag für die Änderung:

`test that users without keycloak-uploader-role can upload documents only if assigned company-owner-role`

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt [hunk 2]
```diff
@@ -91,13 +95,63 @@ class DocumentControllerTest {
...
+        )
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+
+        val clientException = assertThrows<ClientException> {
+            uploadDocument(pdfDocument, TechnicalUser.Reader)
+        }
+        assertEquals("Client error : 403 ", clientException.message)
+    }
+
+    private fun ensureUserIsNotOwnerOfAnyCompany(userId: UUID) {
```
Comment (emanueldfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt:130]: Ich schlage vor dass wir die Funktion leicht umbennen, da sie zunächst den Eindruck erweckt, man würde nur eine Validierung durchführen.
In Wahrheit manipuliert man (potenziell) tatsächlich die DB, indem man role-assignments removet.

Vorschlag von mir wäre z.B. `private fun removePotentialCompanyOwnerRolesFromUser`
oder nur `private fun removeCompanyOwnerRolesFromUser`

Comment (emanueldfine) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt:130]: Oder noch besser `private fun removeAllCompanyOwnershipsFromUser`

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt [hunk 3]
```diff
@@ -91,13 +95,63 @@ class DocumentControllerTest {
...
+
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+        assertDoesNotThrow { uploadDocument(pdfDocument, TechnicalUser.Reader) }
+
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        apiAccessor.companyRolesControllerApi.removeCompanyRole(
+            CompanyRole.CompanyOwner,
+            testCompanyId,
+            dataReaderId,
+        )
```
[the remark this answers is not in the document]
Comment (MichBlank) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt:121]: Done, ein API call mehr gegen besser lesbaren Code getauscht :)

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt (hunk 2 above)
Comment (MichBlank) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt:130]: Done, warum man das macht ist ja auch klar ohne es explizit im Namen haben zu müssen

**Pull request merged**